aviary/utils/engine_deck_conversion.py

```diff
@@ -485,27 +485,35 @@ def _read_map(f, is_turbo_prop=False):
     map_data[:, 0] = amap
 
     # number of points on a single line - wrapped if more than 6
-    nptloc = min(6, npts)
+    max_columns = 6
 
     # point vals: FORMAT(10X,6F10.4,10X)
-    x = list(_parse(f, [(None, 10), *_rep(nptloc, (float, 10))]))
-    if npts > nptloc:
+    x = []
+    npts_remaining = npts
+    while npts_remaining > 0:
+        npts_to_read = min(max_columns, npts_remaining)
         # remaining vals on wrapped line
-        x.extend(list(_parse(f, [(None, 10), *_rep(npts - nptloc, (float, 10))])))
+        x.extend(list(_parse(f, [(None, 10), *_rep(npts_to_read, (float, 10))])))
+        npts_remaining -= npts_to_read
 
     map_data[:, 2] = np.tile(x, nline)
 
     for j in range(nline):
+        npts_remaining = npts
+        npts_to_read = min(max_columns, npts_remaining)
         # line (y) val then z vals: FORMAT(F10.4,6F10.1,10X,/(6F10.1,10X))
-        vals = list(_parse(f, [(float, 10), *_rep(nptloc, (float, 10))]))
+        vals = list(_parse(f, [(float, 10), *_rep(npts_to_read, (float, 10))]))
         y = vals[0]
         z = vals[1:]
-        if npts > nptloc:
+        npts_remaining -= npts_to_read
+        while npts_remaining > 0:
+            npts_to_read = min(6, npts_remaining)
             # add remaining vals on warapped line
-            line_format = [*_rep(npts - nptloc, (float, 10))]
+            line_format = [*_rep(npts_to_read, (float, 10))]
             if is_turbo_prop:
                 line_format = [(None, 10), *line_format]
             z.extend(list(_parse(f, line_format)))
+            npts_remaining -= npts_to_read
```

Comment:
Slightly reduces code duplication and matches the x read more closely?
```
        npts_remaining = npts
        # line (y) val then z vals: FORMAT(F10.4,6F10.1,10X,/(6F10.1,10X))
        first_line = True
        vals = []
        while npts_remaining > 0:
            npts_to_read = min(max_columns, npts_remaining)
            # add remaining vals on wrapped line
            line_format = [*_rep(npts_to_read, (float, 10))]
            if first_line:
                [(float, 10), *line_format]
                first_line = False
            elif is_turbo_prop:
                line_format = [(None, 10), *line_format]
            vals.extend(list(_parse(f, line_format)))
            npts_remaining -= npts_to_read
        y = vals[0]
        z = vals[1:]
```

Reply:
I tried replacing the block with this code but it failed during the engine conversion test. I rolled it back to what's there for now.